src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -315,313 +346,322 @@ void func_80B67540(EnSth* this, PlayState* play) {
                 func_801477B4(play);
-                this->actionFunc = func_80B67B50;
-                func_80B67B50(this, play);
+                this->actionFunc = EnSth_SwampSpiderHouseGiveMask;
+                EnSth_SwampSpiderHouseGiveMask(this, play);
                 break;
 
-            case 0x91A:
+            case 0x91A: // Someone gave me this mask and said it would make me rich, getting rid of it
+                // TODO: why is this two flags? is it some.. started dialogue, ended dialogue flag?
                 SET_WEEKEVENTREG(WEEKEVENTREG_34_40);
-                CLEAR_WEEKEVENTREG(WEEKEVENTREG_34_08);
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_MASK_OF_TRUTH);
 
-            case 0x902:
-            case 0x903:
-            case 0x90E:
-            case 0x90F:
-            case 0x918:
-            case 0x91B:
-                func_80B670A4(this, 3);
+            case 0x902: // (does not exist)
+            case 0x903: // (does not exist)
+            case 0x90E: // (does not exist)
+            case 0x90F: // (does not exist)
+            case 0x918: // I have had enough, going home
+            case 0x91B: // As soon as I calm down, getting rid of it
+                EnSth_SwitchAnimation(this, STH_ANIM_WAIT);
 
             default:
                 this->actor.flags &= ~ACTOR_FLAG_10000;
                 func_801477B4(play);
-                this->actionFunc = func_80B67DA0;
+                this->actionFunc = EnSth_SwampSpiderHouseIdle;
                 break;
         }
     }
 }
 
-void func_80B67DA0(EnSth* this, PlayState* play) {
+void EnSth_SwampSpiderHouseIdle(EnSth* this, PlayState* play) {
     SkelAnime_Update(&this->skelAnime);
 
     if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
-        func_80B67984(this, play);
-        this->actionFunc = func_80B67C1C;
-    } else if (func_80B6703C(this, play)) {
+        EnSth_GetInitialSwampSpiderHouseText(this, play);
+        this->actionFunc = EnSth_HandleSwampSpiderHouseConversation;
+    } else if (EnSth_CanSpeakToPlayer(this, play)) {
         this->actor.textId = 0;
         func_800B8614(&this->actor, play, 110.0f);
     }
 }
 
-void func_80B67E20(Actor* thisx, PlayState* play) {
+void EnSth_OceansideSpiderHouseWaitForTokens(Actor* thisx, PlayState* play) {
     EnSth* this = THIS;
 
-    if (Inventory_GetSkullTokenCount(play->sceneId) >= 30) {
-        this->actor.update = func_80B680A8;
-        this->actor.draw = func_80B6849C;
+    if (Inventory_GetSkullTokenCount(play->sceneId) >= STH_OCEANSIDE_SPIDER_TOKENS_REQUIRED) {
+        this->actor.update = EnSth_Update;
+        this->actor.draw = EnSth_Draw;
         this->actor.flags |= ACTOR_FLAG_1;
     }
 }
 
-void EnSth_Update(Actor* thisx, PlayState* play) {
+/**
+ * Dual object actors have to wait for the object to finish loading,
+ * this dev chose to use a temporary update instead of temporary acitonFunc
+ */
+void EnSth_WaitForObject(Actor* thisx, PlayState* play) {
```

Comment:
Even if it's only a temporary update, I think it should still have `Update` in it's name somehow